packages/flutter/lib/src/gestures/multitap.dart

```diff
@@ -587,3 +587,411 @@ class MultiTapGestureRecognizer extends GestureRecognizer {
+  /// [onSerialTapCancel] nor [onSerialTapUp] have yet fired.
+  bool get isTrackingPointer => _pendingTap != null;
+
+  @override
+  bool isPointerAllowed(PointerDownEvent event) {
+    if (onSerialTapDown == null &&
+        onSerialTapCancel == null &&
+        onSerialTapUp == null) {
+      return false;
+    }
+    return super.isPointerAllowed(event);
+  }
+
+  @override
+  void addAllowedPointer(PointerDownEvent event) {
+    if ((_completedTaps.isNotEmpty && !_representsSameSeries(_completedTaps.last, event))
+        || _pendingTap != null) {
+      _reset();
+    }
+    _trackTap(event);
+  }
+
+  bool _representsSameSeries(_TapTracker tap, PointerDownEvent event) {
+    return tap.hasElapsedMinTime() // touch screens often detect touches intermittently
+        && tap.hasSameButton(event)
+        && tap.isWithinGlobalTolerance(event, kDoubleTapSlop);
+  }
+
+  void _trackTap(PointerDownEvent event) {
+    _stopSerialTapTimer();
+    if (onSerialTapDown != null) {
+      final SerialTapDownDetails details = SerialTapDownDetails(
+        globalPosition: event.position,
+        localPosition: event.localPosition,
+        kind: getKindForPointer(event.pointer),
+        buttons: event.buttons,
+        count: _completedTaps.length + 1,
+      );
+      invokeCallback<void>('onSerialTapDown', () => onSerialTapDown!(details));
+    }
+    final _TapTracker tracker = _TapTracker(
+      event: event,
+      entry: GestureBinding.instance!.gestureArena.add(event.pointer, this),
+      doubleTapMinTime: kDoubleTapMinTime,
+    );
+    assert(_pendingTap == null);
+    _pendingTap = tracker;
+    tracker.startTrackingPointer(_handleEvent, event.transform);
+  }
+
+  void _handleEvent(PointerEvent event) {
+    assert(_pendingTap != null);
+    assert(_pendingTap!.pointer == event.pointer);
+    final _TapTracker tracker = _pendingTap!;
+    if (event is PointerUpEvent) {
+      _registerTap(event, tracker);
+    } else if (event is PointerMoveEvent) {
+      if (!tracker.isWithinGlobalTolerance(event, kDoubleTapTouchSlop)) {
+        _reset();
+      }
+    } else if (event is PointerCancelEvent) {
+      _reset();
+    }
+  }
+
+  @override
+  void acceptGesture(int pointer) {
+    assert(_pendingTap != null);
+    assert(_pendingTap!.pointer == pointer);
+    _acceptedGestures[pointer] = true;
+  }
+
+  @override
+  void rejectGesture(int pointer) {
+    _acceptedGestures[pointer] = false;
+    _reset();
+  }
+
+  void _rejectPendingTap() {
+    assert(_pendingTap != null);
+    final _TapTracker tracker = _pendingTap!;
+    _pendingTap = null;
+    // Order is important here; the `resolve` call can yield a re-entrant
+    // `reset()`, so we need to check cancel here while we can trust the
+    // length of our _completedTaps list.
+    _checkCancel(_completedTaps.length + 1);
+    if (_acceptedGestures[tracker.pointer] != false) {
+      tracker.entry.resolve(GestureDisposition.rejected);
+    }
```

Comment:
This confuses me: If the gesture is accepted (_acceptedGestures[tracker.pointer] is true) we need to reject it? Isn't that too late for rejection?

Reply:
I guess this works because "It's fine to attempt to resolve a gesture recognizer for an arena that is already resolved." (from https://master-api.flutter.dev/flutter/gestures/GestureArenaEntry/resolve.html)

Reply:
But since we keep track, could we not just only reject it if we have to?

Reply:
Yup - changed this to `if (!_gestureResolutions.containsKey(event.pointer)) { ... }`

(also renamed `_acceptedGestures` to be `_gestureResolutions` and made it map to `GestureDisposition` rather than bool for clarity)